lib/widgets/app.dart

```diff
@@ -221,15 +223,63 @@ class ChooseAccountPage extends StatelessWidget {
     required Widget title,
     Widget? subtitle,
   }) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
     return Card(
       clipBehavior: Clip.hardEdge,
       child: ListTile(
         title: title,
         subtitle: subtitle,
+        trailing: PopupMenuButton<AccountItemOverflowMenuItem>(
+          iconColor: designVariables.icon,
+          itemBuilder: (context) => [
+            PopupMenuItem(
+              value: AccountItemOverflowMenuItem.logOut,
+              child: Text(zulipLocalizations.chooseAccountPageLogOutButton)),
+          ],
+          onSelected: (item) async {
+            switch (item) {
+              case AccountItemOverflowMenuItem.logOut: {
+                unawaited(_logOutAccount(context, accountId));
+              }
+            }
+          }),
+        // The default trailing padding with M3 is 24px. Decrease by 12 because
+        // PopupMenuButton adds 12px padding on all sides of the "…" icon.
+        contentPadding: const EdgeInsetsDirectional.only(start: 16, end: 12),
         onTap: () => Navigator.push(context,
           HomePage.buildRoute(accountId: accountId))));
   }
 
+  Future<void> _logOutAccount(BuildContext context, int accountId) async {
+    final globalStore = GlobalStoreWidget.of(context);
+
+    final account = globalStore.getAccount(accountId);
+    if (account == null) return; // TODO(log)
+
+    // Async IIFE to not block removing the account on the unregister-token request.
+    unawaited(() async {
```

Comment:
nit: move this anonymous function to be its own private method with a name; it's 17 lines long, so I think that'll help make the overall structure easier to see